jetty-util/src/main/java/org/eclipse/jetty/util/FutureCallback.java

```diff
@@ -163,9 +163,7 @@ public void block(long timeout, TimeUnit unit) throws IOException
         {
             Throwable cause = e.getCause();
             if (cause instanceof RuntimeException)
-                throw (RuntimeException)cause;
-            else if (cause instanceof IOException)
-                throw (IOException)cause;
+                throw new RuntimeException(cause);
```

Comment:
You don't want to wrap a `RuntimeException` inside another... the old code was correct.

Reply:
What about rewrapping of the `IOException`, I was getting stack traces that were being thrown and they didn't trace back through the`FutureCallback.block()` line and just some seemingly unrelated code, so I thought that was a bit weird and decided to wrap them instead so you could see the block call in the stack trace. I can revert this, wasn't sure what the correct behaviour was.

Reply:
`block() throws IOException`, so it throws anyway, which means that may throw `Error`, `RuntimeException` and `IOException`.
The old code was already doing the right thing: rethrow `RuntimeException` and `IOException`, and wrap the rest.
The only improvement would be to let out `Error` and only wrap checked `Exception` into `IOException`.

Reply:
I see now, leave this rewrapping then, so applications will have a nicer stack trace that get lost because we unwrap `ExecutionException`.